examples/using-contentful/src/templates/product.js

```diff
@@ -86,13 +86,13 @@ export const pageQuery = graphql`
       }
       brand {
         companyName {
-          companyName
+          raw
         }
       }
       categories {
         id
         title {
```

Comment:
This fails as references currently always use the `ContentfulEntry` interface which includes all content types.

Will change the schema to detect reference type validation and use single-type/unions as types for these fields.